app/scripts/controllers/permissions/background-api.test.js

```diff
@@ -1,197 +1,620 @@
+
+      expect(permissionController.updateCaveat).toHaveBeenCalledWith(
         'foo.com',
-        RestrictedMethods.eth_accounts,
-        CaveatTypes.restrictReturnedAccounts,
+        Caip25EndowmentPermissionName,
+        Caip25CaveatType,
+        {
+          requiredScopes: {
+            'eip155:1': {
+              methods: [],
+              notifications: [],
+              accounts: ['eip155:1:0x3', 'eip155:1:0x1'],
+            },
+            'eip155:10': {
+              methods: [],
+              notifications: [],
+              accounts: ['eip155:10:0x3', 'eip155:10:0x1'],
+            },
+          },
+          optionalScopes: {
+            'bip122:000000000019d6689c085ae165831e93': {
+              methods: [],
+              notifications: [],
+              accounts: [
+                'bip122:000000000019d6689c085ae165831e93:128Lkh3S7CkDTBZ8W7BbpsN3YYizJMp8p6',
+              ],
+            },
+            'eip155:1': {
+              methods: [],
+              notifications: [],
+              accounts: ['eip155:1:0x3', 'eip155:1:0x1'],
+            },
+          },
+          isMultichainOrigin: true,
+        },
       );
-
-      expect(permissionController.revokePermission).not.toHaveBeenCalled();
-      expect(permissionController.updateCaveat).not.toHaveBeenCalled();
     });
   });
 
   describe('requestAccountsPermissionWithId', () => {
-    it('request an accounts permission and returns the request id', async () => {
+    it('gets the networkConfiguration for the currently selected network client', () => {
+      const networkController = {
+        state: {
+          selectedNetworkClientId: 'mainnet',
+        },
+        getNetworkConfigurationByNetworkClientId: jest.fn().mockReturnValue({
+          chainId: '0x1',
+        }),
+      };
+      const approvalController = {
+        addAndShowApprovalRequest: jest.fn().mockResolvedValue({
+          approvedChainIds: ['0x1', '0x5'],
+          approvedAccounts: ['0xdeadbeef'],
+        }),
+      };
       const permissionController = {
-        requestPermissions: jest
-          .fn()
-          .mockImplementationOnce(async (_, __, { id }) => {
-            return [null, { id }];
-          }),
+        grantPermissions: jest.fn(),
       };
 
-      const id = await getPermissionBackgroundApiMethods(
+      getPermissionBackgroundApiMethods({
+        networkController,
+        approvalController,
         permissionController,
-      ).requestAccountsPermissionWithId('foo.com');
+      }).requestAccountsPermissionWithId('foo.com');
 
-      expect(permissionController.requestPermissions).toHaveBeenCalledTimes(1);
-      expect(permissionController.requestPermissions).toHaveBeenCalledWith(
-        { origin: 'foo.com' },
-        { eth_accounts: {} },
-        { id: expect.any(String) },
-      );
+      expect(
+        networkController.getNetworkConfigurationByNetworkClientId,
+      ).toHaveBeenCalledWith('mainnet');
+    });
```

Comment:
should we add a check that `permissionController.grantPermissions` is call with expected params?

Reply:
Oh I see you do this check in a subsequent test.